libs/wcm/utils/sessionHandlers.js

```diff
@@ -0,0 +1,65 @@
+import { getSdkError } from '@walletconnect/utils';
+import { to } from 'await-to-js';
+import { client } from '@libs/wcm/utils/connectionCreator';
+import { PAIRING_PROPOSAL_STATUS, ERROR_CASES } from '../constants/lifeCycle';
+
+/**
+ * The approve handler for connection proposal
+ *
+ * @param {object} proposal The proposal object as received via the event
+ * @param {array} selectedAccounts List of lisk addresses selected by the user
+ * @returns {Promise} The promise that resolves when the approval is complete
+ */
+export const onApprove = async (
+  proposal, selectedAccounts,
+) => {
+  const { id, params } = proposal;
+  const { requiredNamespaces, relays } = params;
+
+  // Normalize the information according to requirements of the bridge
+  const namespaces = Object.keys(requiredNamespaces).reduce((nms, key) => {
```

Comment:
I couldn't improve further. Suggestions are welcome.

Reply:
selectedAccounts = ['lskj34x8zh85zh4khjq64ofudmjax2hzc5hxw7vok']

requiredNamespaces = {
  lsk: {
    chains: [
      'Lisk:1',
      'Lisk:2',
    ],
  },
  duEdu: {
    chains: [
      'DueEdu:1',
      'DueEdu:2',
      'DueEdu:3',
      'DueEdu:4',
    ]
  }
}

Reply:
We changed the implementation. The new version seems more readable, although it's 35% slower:
<img width="1228" alt="Screenshot 2022-08-31 at 17 42 13" src="https://user-images.githubusercontent.com/666685/187721272-d792d4a3-abe5-49ce-9a0e-ade1ceced7bd.png">


I'm open to improvements if you have any.

Reply:
this is not a valid test because they not running in sandbox
<img width="616" alt="Screenshot 2022-09-02 at 16 05 46" src="https://user-images.githubusercontent.com/1031663/188166785-82ca5625-f50c-4650-bf7e-e0c1ddbd874a.png">